test/controllers/admin/appointment_detail_controller_test.rb

```diff
@@ -0,0 +1,34 @@
+require "application_system_test_case"
+
+module Admin
+  class AppointmentDetailsControllerTest < ActionDispatch::IntegrationTest
+    include Devise::Test::IntegrationHelpers
+
+    setup do
+      @member = create(:member)
+      @loan = create(:loan, member: @member)
+      @appointment = build(:appointment, member: @member)
+      @appointment.loans << @loan
+      @appointment.save!
+
+      @user = create(:admin_user)
+      sign_in @user
+    end
+
+    test "completes an appointment" do
+      post admin_appointment_detail_path(@appointment), as: :turbo_stream
+
+      assert @appointment.reload.completed_at
+      assert_redirected_to admin_appointments_path
```

Comment:
nit: I would probably rearrange these line so that the assertion related to the response comes first, and then the one about the database state is later. This matched the other new test and our general convention.

```suggestion
      assert_redirected_to admin_appointments_path

      assert @appointment.reload.completed_at
```

Reply:
Reordered assert statements.